### PR TITLE
#299 Implement oEmbed API methods

### LINKED ISSUE #299: Implement OEmbed API
Mastodon's OEmbed API allows a user to generate OEmbed previews. The goal of this task is to implement the following endpoints of this API in BigBone:

- Get OEmbed info as JSON

The OEmbed API as it is documented by Mastodon can be found here:
https://docs.joinmastodon.org/methods/oembed/

Tip: A lot of other endpoints similar to this have already been implemented. In order to get an idea how to implement this, I suggest that you study the code upfront. Start with one of the `*Methods.kt` files.

Please consult the following documents to make sure your contribution meets our projects criteria:

- [Code of Conduct](https://github.com/andregasser/bigbone/blob/master/CODE_OF_CONDUCT.md)
- [Code Contribution Guidelines](https://github.com/andregasser/bigbone/blob/master/CONTRIBUTING.md)

### PR DESCRIPTION
**Pull request by PattaFeuFeu**
# Description

Implements the only currently available method of the [oEmbed API methods](https://docs.joinmastodon.org/methods/oembed/).

Nothing funky going on here, just a pretty straightforward endpoint implementation.

Closes #299

# Type of Change

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update
  - [API Coverage](https://github.com/andregasser/bigbone/wiki/Mastodon-API-Coverage) page needs to be updated

# How Has This Been Tested?

I’ve added two samples and ran `gradle check`

# Mandatory Checklist

- [x] My change follows the projects coding style
- [x] I ran `gradle check` and there were no errors reported
- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

# Optional Things To Check

- [x] In case you worked on new features: Did you also do the Rx part?
- [x] In case you added new *Methods classes: Did you also add it to `MastodonClient`?
- [x] In case you worked on endpoints: Please mention in the PR that [API Coverage](https://github.com/andregasser/bigbone/wiki/Mastodon-API-Coverage) page needs to be updated (if needed)

### CODE REVIEW

File: bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt [hunk 1]
```diff
@@ -213,6 +214,13 @@ private constructor(
     @get:JvmName("oauth")
     val oauth: OAuthMethods by lazy { OAuthMethods(this) }
 
+    /**
+     * Access API methods under "oembed" endpoint.
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt:218]: Just a really minor comment on this KDoc here: We should make these comments consistent over all endpoints listed in here. Sometimes we write **Access API methods under "api/vX/notification[s]" endpoint.** and sometimes it is only **Access API methods under "oembed" endpoint.** I am happy with both as long as we are consistent.

File: bigbone/src/main/kotlin/social/bigbone/api/entity/OEmbedMetadata.kt [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+/**
+ * oEmbed metadata returned by [OEmbedMethods].
+ * @see <a href="https://oembed.com/#section2.3">OEmbed response parameters documentation</a>
+ */
+@Serializable
+data class OEmbedMetadata(
+    /**
+     * The resource type.
+     * It’s unclear from <a href="https://docs.joinmastodon.org/methods/oembed/#get">Mastodon’s documentation</a>
```
Comment (andregasser) [inline bigbone/src/main/kotlin/social/bigbone/api/entity/OEmbedMetadata.kt:15]: Looking at Mastodon's OEmbed Serializer, I can see that **rich** is the only value they deliver at the moment: https://github.com/mastodon/mastodon/blob/main/app/serializers/oembed_serializer.rb.

**Review by andregasser — CHANGES_REQUESTED** [review]
Thanks for implementing it! Just had some minor remarks, otherwise looking good!

File: bigbone/src/main/kotlin/social/bigbone/api/entity/OEmbedMetadata.kt (hunk 2 above)
Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/api/entity/OEmbedMetadata.kt:15]: Ah, thanks for checking! I’ll update the documentation and default value.

File: bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt (hunk 1 above)
Comment (PattaFeuFeu) [reply bigbone/src/main/kotlin/social/bigbone/MastodonClient.kt]: I’ve unified them all. I’ve also sorted them alphabetically again. To make reviewing that easier, I have created separate commits for both.

**Review by andregasser — APPROVED** [review]

**Pull request merged**